src/proto/ctl/storage_nvme.proto

```diff
@@ -65,13 +58,30 @@ message NvmeController {
 		bool volatile_mem_warn = 20; // volatile memory backup
 	}
 
+	// Namespace represents a namespace created on an NvmeController.
+	message Namespace {
+		uint32 id = 1;			// namespace id
+		uint64 size = 2;		// device capacity in bytes
+		string ctrlrpciaddr = 3;	// parent controller PCI address
+	}
+
+	// SmdDevice represents a blobstore created on a NvmeController_Namespace.
+	// TODO: this should really be embedded in Namespace above.
```

Comment:
Why not just do it now, then?

Reply:
more changes would be required to pass the namespace identifier back alongside the Smd info and I didn't want to add further complexity to this PR, that would really be a cosmetic change as we don't really use the namespace info other than calculating cumulative capacity